src/algorithms/uast_to_bag_paths.py

```diff
@@ -0,0 +1,62 @@
+from sourced.ml.algorithms import NoopTokenParser
+from algorithms.path_contexts import get_paths
+from sourced.ml.utils import PickleableLogger
+
+
+class FakeVocabulary:
+    # FIXME(zurk): change to simple function. Vadim Markovtsev comments:
+    # > would rather made this a simple function and change roles2index
+    # type from [] to callable. Saves time to understand.
+    def __getitem__(self, item):
+        return item
+
+
+class Uast2BagOfPaths(PickleableLogger):
```

Comment:
if it is a class related to extraction algorithm you should put all algorithm code inside this class. 
And maybe `ExtNode` class with code related to Node processing.